authorizations/authorization-chain/src/main/java/org/apache/gravitino/authorization/chain/ChainAuthorizationPlugin.java

```diff
@@ -0,0 +1,283 @@
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.gravitino.Catalog;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.authorization.AuthorizationProperties;
+import org.apache.gravitino.authorization.ChainAuthorizationProperties;
+import org.apache.gravitino.authorization.Group;
+import org.apache.gravitino.authorization.MetadataObjectChange;
+import org.apache.gravitino.authorization.Owner;
+import org.apache.gravitino.authorization.Role;
+import org.apache.gravitino.authorization.RoleChange;
+import org.apache.gravitino.authorization.User;
+import org.apache.gravitino.connector.authorization.AuthorizationPlugin;
+import org.apache.gravitino.connector.authorization.BaseAuthorization;
+import org.apache.gravitino.exceptions.AuthorizationPluginException;
+import org.apache.gravitino.utils.IsolatedClassLoader;
+
+/** Chain authorization operations plugin class. <br> */
+public class ChainAuthorizationPlugin implements AuthorizationPlugin {
+  private List<AuthorizationPlugin> plugins = Lists.newArrayList();
+  private final String metalake;
+
+  public ChainAuthorizationPlugin(
+      String metalake, String catalogProvider, Map<String, String> config) {
+    this.metalake = metalake;
+    initPlugins(catalogProvider, config);
+  }
+
+  private void initPlugins(String catalogProvider, Map<String, String> properties) {
+    ChainAuthorizationProperties chainAuthorizationProperties =
+        new ChainAuthorizationProperties(properties);
+    chainAuthorizationProperties.validate();
+    // Validate the properties for each plugin
+    chainAuthorizationProperties
+        .plugins()
+        .forEach(
+            pluginName -> {
+              Map<String, String> pluginProperties =
+                  chainAuthorizationProperties.fetchAuthPluginProperties(pluginName);
+              String authProvider = chainAuthorizationProperties.getPluginProvider(pluginName);
+              AuthorizationProperties.validate(authProvider, pluginProperties);
+            });
+    // Create the plugins
+    chainAuthorizationProperties
+        .plugins()
+        .forEach(
+            pluginName -> {
+              String authProvider = chainAuthorizationProperties.getPluginProvider(pluginName);
+              Map<String, String> pluginConfig =
+                  chainAuthorizationProperties.fetchAuthPluginProperties(pluginName);
+
+              ArrayList<String> libAndResourcesPaths = Lists.newArrayList();
+              BaseAuthorization.buildAuthorizationPkgPath(
+                      ImmutableMap.of(Catalog.AUTHORIZATION_PROVIDER, authProvider))
+                  .ifPresent(libAndResourcesPaths::add);
+              IsolatedClassLoader classLoader =
+                  IsolatedClassLoader.buildClassLoader(libAndResourcesPaths);
```

Comment:
Why do we need to build different classloaders for different plugins, can we just use catalog's classloader?

Reply:
When we load `ChainAuthorizationPlugin` , We need add `authorizaitons/chain/libs` in the `BaseCatalog` IsolatedClassLoader class paths.
We cann't add `authorizaitons/ranger/libs`  at same time. So we need an IsolatedClassLoader to separate load `RangerAuthorizationPlugin` here.